pkg/controller/config/config_controller.go

```diff
@@ -251,13 +251,17 @@ func (r *ReconcileConfig) Reconcile(request reconcile.Request) (reconcile.Result
 
 	// If the watch set has not changed, we're done here.
 	if r.watched.Equals(newSyncOnly) && r.processExcluder.Equals(newExcluder) {
-		if !r.wiped && r.needsReplay != nil {
+		// ...unless we have pending wipe / replay operations from a previous reconcile.
+		if !(r.needsWipe || r.needsReplay != nil) {
 			return reconcile.Result{}, nil
 		}
+
+		// If we reach here, the watch set hasn't changed since last reconcile, but we
+		// have unfinished wipe/replay business from the last change.
 	} else {
-		// only re-calculate replays if our watch set has changed
+		// The watch set _has_ changed, so recalculate the replay set.
 		r.needsReplay = nil
-		r.wiped = false
+		r.needsWipe = true
```

Comment:
Yep, was still getting the conditions wrong. Much easier to reason about things with the new names, thanks!